core/trino-main/src/test/java/io/trino/sql/TestExpressionInterpreter.java

```diff
@@ -284,6 +285,27 @@ public void testIsNotNull()
         assertOptimizedEquals("bound_decimal_long IS NOT NULL", "true");
     }
 
+    @Test
+    public void testLambdaBody()
+    {
+        assertOptimizedEquals("\"" + TryFunction.NAME + "\"(() -> 100 * 100)",
+                "\"" + TryFunction.NAME + "\"(() -> 10000)");
+        assertOptimizedEquals("\"" + TryFunction.NAME + "\"(() -> CAST('123' as BIGINT))",
+                "\"" + TryFunction.NAME + "\"(() -> BIGINT '123')");
+        assertOptimizedEquals("\"" + TryFunction.NAME + "\"(() -> CAST('abc' as BIGINT))",
+                "\"" + TryFunction.NAME + "\"(() -> CAST('abc' as BIGINT))");
+        assertOptimizedEquals("\"" + TryFunction.NAME + "\"(() -> IF(false, 1, 0 / 0))",
+                "\"" + TryFunction.NAME + "\"(() -> 0 / 0)");
+        assertOptimizedEquals("\"" + TryFunction.NAME + "\"(() -> 5 / 0)",
+                "\"" + TryFunction.NAME + "\"(() -> 5 / 0)");
+        assertOptimizedEquals("\"" + TryFunction.NAME + "\"(() -> nullif(true, true))",
+                "\"" + TryFunction.NAME + "\"(() -> CAST(null AS Boolean))");
```

Comment:
Rewrite these tests to use a function other that TryFunction.NAME, which is an internal implementation detail and not meant to be used directly. We don't want to introduce dependencies on it.

Reply:
Good catch, I replaced `TryFunction.NAME` with `transform`.